app/controllers/claims_controller.rb

```diff
@@ -68,13 +66,16 @@ def show
   end
 
   def update
+    params[:claim][:hmrc_validation_attempt_count] = session[:hmrc_validation_attempt_count] || 0 if on_banking_page?
+
     # TODO: Migrate the remaining slugs to form objects.
     if (@form = journey.form(claim: current_claim, params:))
       if @form.save
         retrieve_student_loan_details
         update_session_with_selected_policy
         redirect_to claim_path(current_journey_routing_name, next_slug)
       else
+        session[:hmrc_validation_attempt_count] = (session[:hmrc_validation_attempt_count] || 0) + 1 if on_banking_page? && @form.hmrc_api_validation_attempted?
```

Comment:
Could we store `hmrc_validation_attempt_count` as a column on the `Claim`? That should let us keep the attempt counting logic contained in the form object? Should be safe to just set a column default of `0` for existing claims and as the app isn't live for new claims I don't think we need to worry about handling inflight claims

Reply:
thanks Rich! this is a nice idea and would keep the logic inside the form. i think i'll leave it in the session for now since it's mirroring the current behaviour and isn't not something we need to know about once the user has submitted a claim.